pkg/flow/internal/controller/loader.go

```diff
@@ -391,14 +425,14 @@ func (l *Loader) evaluate(logger log.Logger, parent *vm.Scope, c *ComponentNode)
 
 // evaluateConfig constructs the final context for the special config Node and
 // evaluates it. mut must be held when calling evaluateConfig.
-func (l *Loader) evaluateConfig(logger log.Logger, parent *vm.Scope, c *ConfigNode) (*ast.BlockStmt, error) {
+func (l *Loader) evaluateConfigBlock(logger log.Logger, parent *vm.Scope, c BlockNode) error {
 	ectx := l.cache.BuildContext(parent)
-	errBlock, err := c.Evaluate(ectx)
+	err := c.Evaluate(ectx)
 	if err != nil {
 		level.Error(logger).Log("msg", "failed to evaluate config", "node", c.NodeID(), "err", err)
-		return errBlock, err
+		return err
 	}
-	return nil, nil
+	return nil
 }
```

Comment:
IMO it's a bit awkward that the function is called `evaluateConfigBlock` but it takes any `BlockNode`. Moving the calls to CacheArguments/CacheExports in evaluate out to a new function and refactoring a bit so we don't need two separate `evaluate*` methods?

Reply:
I agree in principle and will need slightly longer to digest and come up with an implementation for this 1 (after the community call probably)

Reply:
this should be improved